### PR TITLE
Improve user settings design

### PR DESCRIPTION
**Pull request by joaomilho**
I was annoyed by how hard it was to find which switch related to which item in the feature flags list. Maybe I have bad spacial coordination, or bad vision. But maybe the UX was just bad. Anyways, my suggestion here is 1) align the sliders left, since we're mainly working on desktops 2) make the rows buttonish (hover changes color) 3) adds a little divider per line 4) allows clicking on row.

Before:
Screenshot 2021-07-21 at 16 32 49

After:
Screenshot 2021-07-21 at 16 43 14
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/user-settings/src/components/FeatureFlags/FeatureFlagsItem.tsx [hunk 1]
```diff
@@ -31,20 +31,17 @@ type Props = {
...
-        <Switch
-          color="primary"
-          checked={enabled}
-          onChange={() => toggleHandler(flag.name)}
-          name={flag.name}
-        />
-      </Tooltip>
-    </ListItemSecondaryAction>
-  </ListItem>
+  <>
```
Comment (benjdlambert) [inline plugins/user-settings/src/components/FeatureFlags/FeatureFlagsItem.tsx:34]: Don't think these need to be wrapped in a fragment right? Can just be `<ListItem>` as the root

**Comment by timbonicus** [comment]
Interesting notes in the Material Design spec (emphasis mine):

- [Switches](https://material.io/components/switches) are the preferred way to adjust settings on **mobile**
- [Checkboxes](https://material.io/components/checkboxes) are the preferred way to turn an item on or off in a **desktop environment**

So probably we should change to checkboxes here instead of moving the switches to the left.

File: plugins/user-settings/src/components/FeatureFlags/FeatureFlagsItem.tsx (hunk 1 above)
Comment (joaomilho) [reply plugins/user-settings/src/components/FeatureFlags/FeatureFlagsItem.tsx:34]: Right, my bad I was using a <Divider> earlier!

**Comment by joaomilho** [comment]
> Interesting notes in the Material Design spec (emphasis mine):
> 
> * [Switches](https://material.io/components/switches) are the preferred way to adjust settings on **mobile**
> * [Checkboxes](https://material.io/components/checkboxes) are the preferred way to turn an item on or off in a **desktop environment**
> 
> So probably we should change to checkboxes here instead of moving the switches to the left.

I can go for checkboxes, now, on moving back to the right defeats the whole purpose.

On another note, I feel the switch/checkbox guideline is a product of many years ago when people were not that used to switches and I think that's bygones by now.

**Comment by joaomilho** [comment]
Also: the switch changes the state, it saves the preference. People aren't very used to checkboxes being "live".

**Comment by joaomilho** [comment]
A final comment, perhaps we should consider namespacing/grouping flags. In backstage internal there are too many and they are thematic, so a design with a breakdown would be way more user friendly. WDYT?

**Comment by benjdlambert** [comment]
in order to not block the release on this, and to get something out in this release, let's merge as it is, but follow up with moving these to sections at least and checkboxes if possible?

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**